rewrite-maven/src/main/java/org/openrewrite/maven/UpdateMavenModel.java

```diff
@@ -38,15 +38,16 @@ public Xml visitDocument(Xml.Document document, P p) {
 
         MavenResolutionResult resolutionResult = getResolutionResult();
         Pom requested = resolutionResult.getPom().getRequested();
+        requested.getProperties().clear();
 
         Optional<Xml.Tag> properties = document.getRoot().getChild("properties");
-        requested.getProperties().clear();
         if (properties.isPresent()) {
             for (final Xml.Tag propertyTag : properties.get().getChildren()) {
                 requested.getProperties().put(propertyTag.getName(),
                         propertyTag.getValue().orElse(""));
             }
         }
+        requested.getProperties().putAll(resolutionResult.getUserProperties());
```

Comment:
My thinking here is that user properties should override whatever is defined in pom files.

Reply:
Well, I am in doubt... I think you can make the case for both this and the other way around. Let's talk about this!

Reply:
This looks to confirm that command line overrides pom.
https://stackoverflow.com/a/13709976